src/libs/sdlcd/linux/SDL_syscdrom.c

```diff
@@ -195,22 +201,32 @@ static void CheckMounts(const char *mtab)
 	mntfp = setmntent(mtab, "r");
 	if ( mntfp != NULL ) {
 		char *tmp;
-		char *mnt_type;
 		size_t mnt_type_len;
-		char *mnt_dev;
 		size_t mnt_dev_len;
+		char mnt_type_str[STRMAXLEN+1];
+		char mnt_dev_str[STRMAXLEN+1];
+		char *mnt_type = mnt_type_str;
+		char *mnt_dev = mnt_dev_str;
 
 		while ( (mntent=getmntent(mntfp)) != NULL ) {
+			*mnt_type = '\0';
+			*mnt_dev = '\0';
 			mnt_type_len = SDL_strlen(mntent->mnt_type) + 1;
-			mnt_type = SDL_stack_alloc(char, mnt_type_len);
-			if (mnt_type == NULL)
-				continue;  /* maybe you'll get lucky next time. */
+			if (mnt_type_len>STRMAXLEN) {
+				strncpy(mnt_type_str, mntent->mnt_type, STRMAXLEN);
+				mnt_type_str[STRMAXLEN] = '\0';
+			} else {
+				strncpy(mnt_type_str, mntent->mnt_type, mnt_type_len);
+				mnt_type_str[mnt_type_len] = '\0';
+			}
 
 			mnt_dev_len = SDL_strlen(mntent->mnt_fsname) + 1;
-			mnt_dev = SDL_stack_alloc(char, mnt_dev_len);
-			if (mnt_dev == NULL) {
-				SDL_stack_free(mnt_type);
-				continue;
+			if (mnt_dev_len>STRMAXLEN) {
+				strncpy(mnt_dev_str, mntent->mnt_fsname, STRMAXLEN);
+				mnt_dev_str[STRMAXLEN] = '\0';
+			} else {
+				strncpy(mnt_dev_str, mntent->mnt_fsname, mnt_dev_len);
+				mnt_dev_str[mnt_dev_len] = '\0';
```

Comment:
There's a fair bit of code to copy two strings. A couple options are `snprintf` and `strlcpy`.

We can replace all of this:
```c
			*mnt_type = '\0';
			*mnt_dev = '\0';
			mnt_type_len = SDL_strlen(mntent->mnt_type) + 1;
			if (mnt_type_len>STRMAXLEN) {
				strncpy(mnt_type_str, mntent->mnt_type, STRMAXLEN);
				mnt_type_str[STRMAXLEN] = '\0';
			} else {
				strncpy(mnt_type_str, mntent->mnt_type, mnt_type_len);
				mnt_type_str[mnt_type_len] = '\0';
			}
			mnt_dev_len = SDL_strlen(mntent->mnt_fsname) + 1;
			if (mnt_dev_len>STRMAXLEN) {
				strncpy(mnt_dev_str, mntent->mnt_fsname, STRMAXLEN);
				mnt_dev_str[STRMAXLEN] = '\0';
			} else {
				strncpy(mnt_dev_str, mntent->mnt_fsname, mnt_dev_len);
				mnt_dev_str[mnt_dev_len] = '\0';
			}
```

With:
``` c
			snprintf(mnt_type_str, STRMAXLEN, "%s", mntent->mnt_type);
			strncpy(mnt_dev_str, STRMAXLEN, "%s", mntent->mnt_fsname);
```

It null-terminates the destination no matter what, even if the source string is empty (so the pre and post `= '\0'` can be avoided too).  It also won't write beyond the destination length. 

There's also `strlcpy`, which might be even simpler; but both are excellent replacements in this case.
 - https://news.ycombinator.com/item?id=17248446
 - https://www.geeksforgeeks.org/why-strcpy-and-strncpy-are-not-safe-to-use/

Reply:
Thanks for the suggestion! The said code is much shorter now.